ERCS/erc-6774.md

```diff
@@ -0,0 +1,254 @@
+
+##### `transferFor`
+
+Verifies signature following the [EIP-712](./eip-712.md) procedure, increases the nonce of `data.owner` and transfers the `data.bid.tokenId` to `to` account.
+
+This function MUST be called only by authorized NST contracts, the call MUST occur when users are calling the `barter` function.
+
+```solidity
+function transferFor(BarterTerms memory data, address to, bytes memory signature) external;
+```
+
+##### `barter`
+
+Calls the `transferFor` function on the `data.bid.tokenAddr`, MAY be self, and transfers `data.bid.tokenId` to `data.owner`
+
+This function MUST check if `data.bid.tokenAddr` is an authorized contract before calling `transferFor`
+
+```solidity
+function barter(BarterTerms memory data, bytes memory signature) external;
+```
+
+##### `nonce`
+
+Returns the counter state of successful barters concluded as bidder/proposer.
+
+This value MUST be included in the barter terms signed message to prevent a signature from being used multiple times.
+
+```solidity
+function nonce(address account) external view returns (uint256);
+```
+
+##### `isBarterable`
+
+OPTIONAL - Returns `true` is the `tokenAddr` is authorized to perform barter.
+
+```solidity
+function isBarterable(address tokenAddr) external view returns (bool);
+```
+
+#### Events
+
+##### `BarterNetworkUpdated`
+
+MUST trigger when NST contract address is authorized or revoked.
+
+```solidity
+event BarterNetworkUpdated(address indexed tokenAddr, bool indexed barterable);
+```
+
+### Define the barter terms
+
+Barter terms are represented as the `barter` function's arguments and MUST:
+
+- be signed following the [EIP-712](./eip-712.md) procedure
+- be shared equally between NSTs
+- have included for each tokens:
+  - contract address
+  - token ID
+- have included for the signer:
+  - a nonce to prevent signature reuse
+  - address of the signer
+  - a timestamp deadline
+
+Barter terms MAY be represented as nested struct as follows:
+
+```solidity
+struct Component {
+    address tokenAddr;
+    uint256 tokenId;
+}
+
+struct Message {
+    uint256 nonce;
+    address owner;
+    uint48 deadline;
+}
+
+struct BarterTerms {
+    Component bid;
+    Component ask;
+    Message message;
+}
+```
+
+Barters MAY be declined into several types, if any, `barter` and `transferFor` MUST leverage Solidity's functions overloading to implement the specific logic for each barter type. See following example:
+
+```solidity
+{...}
+struct MultiComponent {
+    address tokenAddr;
+    uint256[] tokenIds;
+}
+
+struct MultiBarterTerms {
+    MultiComponent bid;
+    MultiComponent ask;
+    Message message;
+}
+
+function barter(BarterTerms memory data, bytes memory signature) external;
+function barter(MultiBarterTerms memory data, bytes memory signature) external;
+```
+
+### Contract interface
+
+Every contract compliant with this EIP MUST implement [ERC-721](./erc-721.md) where the following methods MUST be leaved unimplemented:
```

Comment:
What does "unimplemented" mean here? Revert if called? If it literally means don't implement the functions from the interface in ERC-721, you aren't really compatible with that standard.